nav2_costmap_2d/plugins/voxel_layer.cpp

```diff
@@ -347,6 +348,41 @@ void VoxelLayer::raytraceFreespace(
     double wpz = *iter_z;
 
     double distance = dist(ox, oy, oz, wpx, wpy, wpz);
+    // If distance to the point in the cloud is lesser than minimum range from which ray tracing is possible do not perform any further computation
```

Comment:
I think you can do the same here as I suggested in the obstacle layer, but now with the voxel grid. Give the min range to `clearVoxelLineInMap` and `updateRaytraceBounds` and update once it itself computes the distances to return if too low.